models/classes/runner/RunnerParamParserTrait.php

```diff
@@ -162,7 +162,8 @@ protected function saveItemResponses($emptyAllowed = true)
                 ? json_decode($this->getRequestParameter('itemResponse'), true)
                 : null;
 
-            if ($serviceContext->getCurrentAssessmentItemRef()->getIdentifier() !== $itemDefinition) {
+            $currentAssessmentItemRef = $serviceContext->getCurrentAssessmentItemRef();
+            if ($currentAssessmentItemRef && $serviceContext->getCurrentAssessmentItemRef()->getIdentifier() !== $itemDefinition) {
```

Comment:
I don't like this solution :( it seems that now everyone can change or add responses after the test finished...

Reply:
the check of session state is done earlier, at time of retrieving of testContext:
https://github.com/oat-sa/extension-tao-testqti/blob/develop/models/classes/runner/QtiRunnerService.php#L1322

This `QtiRunnerItemResponseException` was added only to make sure that item response corresponds current item id (we had a case when response was submitted for wrong item.

This fix fixed the case when session was closed after `\oat\taoQtiTest\models\runner\QtiRunnerService::check()` called.

Reply:
Understood, thank you